packages/edit-post/src/components/header/fullscreen-mode-close/index.js

```diff
@@ -18,21 +18,24 @@ import { wordpress } from '@wordpress/icons';
 import { store as editorStore } from '@wordpress/editor';
 import { store as coreStore } from '@wordpress/core-data';
 import { useReducedMotion } from '@wordpress/compose';
+import { useCallback } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { store as editPostStore } from '../../../store';
 
 function FullscreenModeClose( { showTooltip, icon, href } ) {
-	const { isActive, isRequestingSiteIcon, postType, siteIconUrl } = useSelect(
-		( select ) => {
-			const { getCurrentPostType } = select( editorStore );
+	const { isActive, isRequestingSiteIcon, postType, siteIconUrl, goBack } =
+		useSelect( ( select ) => {
+			const { getCurrentPostType, getEditorSettings } =
+				select( editorStore );
 			const { isFeatureActive } = select( editPostStore );
 			const { getEntityRecord, getPostType, isResolving } =
 				select( coreStore );
 			const siteData =
 				getEntityRecord( 'root', '__unstableBase', undefined ) || {};
+			const _goBack = getEditorSettings()?.goBack;
```

Comment:
I tested a few things before arriving at the `goBack` approach. I decided to go with this initially because it matches the template editing flow and the way the document header bar works.

It does introduce new functionality to the fullscreen WP logo however in that it re-enters  post/page editing mode and doesn't return you to the post/page list immediately.

To get that working I tested storing the postType in `useState` and only updating if `postType.slug !== `wp_template`. That worked fine. 

Also, if it's possible to get the post object/post type via the postId from the URL param, then that could be another option, but `getEntityRecord` requires a `postType` and something like `post.php?post=2` could refer to a page or a post.

Open to suggestions here.

Reply:
I don't mind this personally! To me it feels like backing out of the mode. Conceptually that's slightly different from going "up" the hierarchy, but still feels pretty good to me in practice:

https://github.com/WordPress/gutenberg/assets/14988353/8112536b-2e78-4b4e-8e43-ea81071916c8

I'd be up for this change, personally, it definitely beats being taken to a screen where you can't do anything!